src/simpledb/query/Term.java

```diff
@@ -5,62 +5,95 @@
 
 /**
  * A term is a comparison between two expressions.
+ * 
  * @author Edward Sciore
- *
  */
 public class Term {
    private Expression lhs, rhs;
    private Operator op;
-   
+
    /**
-    * Create a new term that compares two expressions
-    * for equality.
-    * @param lhs  the LHS expression
-    * @param rhs  the RHS expression
+    * Create a new term that compares two expressions for equality.
+    * 
+    * @param lhs the LHS expression
+    * @param rhs the RHS expression
     */
    public Term(Expression lhs, Expression rhs, Operator op) {
       this.lhs = lhs;
       this.rhs = rhs;
       this.op = op;
    }
-   
+
    /**
-    * Return true if both of the term's expressions
-    * evaluate to the same constant,
+    * Return true if both of the term's expressions evaluate to the same constant,
     * with respect to the specified scan.
+    * 
     * @param s the scan
     * @return true if both expressions have the same value in the scan
     */
    public boolean isSatisfied(Scan s) {
       Constant lhsval = lhs.evaluate(s);
       Constant rhsval = rhs.evaluate(s);
+      return isTermSatisfied(lhsval, rhsval);
+   }
+
+   /**
+    * Return true if the term's expressions belong to 2 separate scans and evaluate
+    * to the same constant, in their respective scans.
+    * 
+    * @param s1 the 1st scan
+    * @param s2 the 2nd scan
+    * @return true if both expressions have the same value in the scan
+    */
+   public boolean isSatisfied(Scan s1, Scan s2) {
+      Constant lhsval, rhsval;
+      if (s1.hasField(lhs.asFieldName())) {
+         lhsval = lhs.evaluate(s1);
+         rhsval = rhs.evaluate(s2);
+      } else {
+         lhsval = lhs.evaluate(s2);
+         rhsval = rhs.evaluate(s1);
+      }
+      return isTermSatisfied(lhsval, rhsval);
+   }
+
+   /**
+    * Helper method evaluating lhs and rhs constants with the term's opr
+    * 
+    * @param lhsval lhs constant
+    * @param rhsval rhs constant
+    * @return
+    */
+   private boolean isTermSatisfied(Constant lhsval, Constant rhsval) {
       String opval = op.getVal();
       switch (opval) {
-         case "=":
-            return rhsval.equals(lhsval);
-         case "!=":
-            return !rhsval.equals(lhsval);
-         case "<>":
-            return !rhsval.equals(lhsval);
-         case "<":
-            return lhsval.compareTo(rhsval) < 0;
-         case ">":
-            return lhsval.compareTo(rhsval) > 0;
-         case "<=":
-            return lhsval.compareTo(rhsval) <= 0;
-         case ">=":
-            return lhsval.compareTo(rhsval) >= 0;
-         default:
-            throw new RuntimeException("Unknown term: " + lhsval + " " + opval + " " + rhsval);
+      case "=":
+         return rhsval.equals(lhsval);
+      case "!=":

```

Comment:
need to add in `return !rhsval.equals(lhsval);` ?

Reply:
In the case statement, it means != and <> will execute the same clause, so there is no need to repeat the return statement